Added utility functions to the new NavigationServer:

AndreaCatania opened the pull request:
Re-added functions, to the `NavigationServer`:
```
- Vector3 get_closest_point_to_segment(const Vector3 &p_from, const Vector3 &p_to, const bool &p_use_collision = false);
- Vector3 get_closest_point(const Vector3 &p_point);
- Vector3 get_closest_point_normal(const Vector3 &p_point);
- Object *get_closest_point_owner(const Vector3 &p_point);
```

Hey @mourning20s check this.

mourning20s commented:
@AndreaCatania I tried it out and it seems to work as expected. Once again, thank you for taking the time to do this.

akien-mga commented:
Could you add the relevant documentation too?

AndreaCatania commented:
> Could you add the relevant documentation too?

Done

AndreaCatania commented:
@akien-mga any news about this?

akien-mga commented on doc/classes/Navigation.xml:
```diff
@@ -28,6 +28,46 @@
+			<argument index="0" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the normal for the point returned by [code]get_closest_point[/code].
```
Use `[method get_closest_point]` instead to generate a hyperlink. Same below.

AndreaCatania replied:
done

akien-mga commented on doc/classes/Navigation.xml:
```diff
@@ -28,6 +28,46 @@
+			<argument index="0" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
```
"nearest point ... nearest to the submitted point" sounds unclear.

akien-mga replied:
Maybe "Returns the point closest to the provided [code]point[/code] on the navigation mesh surface."

AndreaCatania replied:
done

akien-mga commented on doc/classes/Navigation2DServer.xml:
```diff
@@ -188,6 +188,28 @@
+			<argument index="1" name="point" type="Vector2">
+			</argument>
+			<description>
+				Returns the owner region RID for the point returned by [code]get_closest_point[/code].
```
Use `[method map_get_closest_point]`.

AndreaCatania replied:
done

akien-mga commented on doc/classes/Navigation2DServer.xml:
```diff
@@ -188,6 +188,28 @@
+			<argument index="1" name="point" type="Vector2">
+			</argument>
+			<description>
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
```
Same comment as for the `Navigation2D` equivalent.

AndreaCatania replied:
done

akien-mga commented on doc/classes/NavigationServer.xml:
```diff
@@ -188,6 +188,54 @@
+			<argument index="1" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the nearest point on the navigation mesh surface nearest to the submitted point.
```
Same as above.

AndreaCatania replied:
done

akien-mga commented on doc/classes/NavigationServer.xml:
```diff
@@ -188,6 +188,54 @@
+			<argument index="1" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the normal for the point returned by [code]get_closest_point[/code].
```
`[method map_get_closest_point]`.

AndreaCatania replied:
done

akien-mga commented on doc/classes/NavigationServer.xml:
```diff
@@ -188,6 +188,54 @@
+			<argument index="1" name="point" type="Vector3">
+			</argument>
+			<description>
+				Returns the owner region RID for the point returned by [code]get_closest_point[/code].
```
`[method map_get_closest_point]`.

AndreaCatania replied:
done

AndreaCatania committed:
Added utility functions to the new NavigationServer:
- Vector3 get_closest_point_to_segment(const Vector3 &p_from, const Vector3 &p_to, const bool &p_use_collision = false);
- Vector3 get_closest_point(const Vector3 &p_point);
- Vector3 get_closest_point_normal(const Vector3 &p_point);
- Object *get_closest_point_owner(const Vector3 &p_point);

akien-mga commented:
Thanks!
